tests/ui/features/lib/TextEditorPage.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+* ownCloud
+*
+* @author Phillip Davis
+* @copyright 2017 Phillip Davis info@individual-it.net
+*
+* This library is free software; you can redistribute it and/or
+* modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or any later version.
+*
+* This library is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+*
+* You should have received a copy of the GNU Affero General Public
+* License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+namespace Page;
+
+use SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException;
+use WebDriver\Key;
+
+class TextEditorPage extends FilesPage
+{
+	protected $newTextFileButtonXpath = './/div[contains(@class, "newFileMenu")]//a[@data-templatename="New text file.txt"]';
+	protected $newTextFileNameInputLabel = 'New text file.txt';
+	protected $newTextFileNameInputXpath = ".//div[contains(@class, \"newFileMenu\")]//a[@data-templatename=\"New text file.txt\"]//input";
+	protected $textFileEditXpath = "//textarea[contains(@class,'ace_text-input')]";
+	protected $textEditorCloseButtonId = "editor_close";
+
+	/**
+	 * type in the field that matches the given xpath and press enter.
+	 * Note: this depends on methods that might only be in the Selenium implementation
+	 *
+	 * @param string $xpath
+	 * @param string $text
+	 * @throws \SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException
+	 */
+	public function typeInFieldAndPressEnter($xpath, $text)
+	{
+		$element = $this->getSession()->getDriver()->getWebDriverSession()->element("xpath", $xpath);
```

Comment:
`getSession()` is Deprecated. Accessing the session from the element is deprecated as of 1.6 and will be impossible in 2.0.
In other places where we need the Session, we do pass it in from the context https://github.com/owncloud/core/blob/master/tests/ui/features/bootstrap/FilesContext.php#L48